client/app/pages/data-sources/DataSourcesList.jsx

```diff
@@ -156,28 +155,14 @@ const DataSourcesListPage = wrapSettingsTab(
 );
 
 export default [
-  {
+  routeWithUserSession({
     path: "/data_sources",
     title: "Data Sources",
-    render: currentRoute => (
-      <AuthenticatedPageWrapper key={currentRoute.key}>
-        <ErrorBoundaryContext.Consumer>
-          {({ handleError }) => <DataSourcesListPage {...currentRoute.routeParams} onError={handleError} />}
-        </ErrorBoundaryContext.Consumer>
-      </AuthenticatedPageWrapper>
-    ),
-  },
-  {
+    render: pageProps => <DataSourcesListPage {...pageProps} />,
```

Comment:
Looks cleaner now, one last thing that went through my mind: how about allow both a Component and a Function in the `render`? This way you can make `render: pageProps => <Component {...pageProps} />` be the default behavior and still can use a function when you need sth different.

Reply:
I don't want to make things too complicated. When having both `render` and `component` we should now define and document which takes precedence when both are specified. Considering that we don't define routes that often and that `render: pageProps => <Component {...pageProps} />` looks clean enough - I don't think we need one more abstraction (feature?) there.

Reply:
I actually meant to just allow both options (a function or a component) to passed under the `render` prop. But it falls under the "one more abstraction" anyway.

Reply:
Ah, I see your idea now. Well, while it's possible, I think it's better if we'll define all routes in a single manner - as a function, to not confuse future us :grin: